From ceb1acd1ed72958895e40d131c0e6ee74464bb77 Mon Sep 17 00:00:00 2001
From: Fabio Erculiani <lxnay@sabayon.org>
Date: Sun, 11 Nov 2012 10:13:03 +0100
Subject: [PATCH] Unquote ${ZPOOL_FORCE} variable usage.

The default value of ${ZPOOL_USAGE} is unset, quoting it would result into
a "" argument passed to zpool. The workaround to this is to use "dozfs=force"
boot argument.
---
 defaults/initrd.scripts | 6 +++---
 1 file modificato, 3 inserzioni(+), 3 rimozioni(-)

diff --git a/defaults/initrd.scripts b/defaults/initrd.scripts
index 7cb41b3..2e8e744 100755
--- a/defaults/initrd.scripts
+++ b/defaults/initrd.scripts
@@ -707,7 +707,7 @@ startVolumes() {
 		then
 			good_msg "Importing ZFS pools"
 
-			/sbin/zpool import -N -a "${ZPOOL_FORCE}"
+			/sbin/zpool import -N -a ${ZPOOL_FORCE}
 
 			if [ "$?" = '0' ]
 			then
@@ -725,12 +725,12 @@ startVolumes() {
 				then
 					good_msg "LUKS detected. Reimporting ${ZFS_POOL}"
 					/sbin/zpool export -f "${ZFS_POOL}"
-					/sbin/zpool import -N "${ZPOOL_FORCE}" "${ZFS_POOL}"
+					/sbin/zpool import -N ${ZPOOL_FORCE} "${ZFS_POOL}"
 				fi
 			else
 				good_msg "Importing ZFS pool ${ZFS_POOL}"
 
-				/sbin/zpool import -N "${ZPOOL_FORCE}" "${ZFS_POOL}"
+				/sbin/zpool import -N ${ZPOOL_FORCE} "${ZFS_POOL}"
 
 				if [ "$?" = '0' ]
 				then
-- 
1.7.12

